### PR TITLE
fix(reveal.js): bug fixes

### PR DESCRIPTION
**Pull request by Mister-Hope**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes.

---

- Improve `slideNumber` option

  The function only has one parameter representing `slideObject`

  TODO: May implement `slideObject` in future PRs

- Update `keyboardCondition` option

  See <https://github.com/hakimel/reveal.js/blob/790fd8c29acba9269fb8105ca13c0620e2bc1474/js/controllers/keyboard.js#L146>

- Update `autoSlideMethod` option

  See <https://github.com/hakimel/reveal.js/blob/790fd8c29acba9269fb8105ca13c0620e2bc1474/js/reveal.js#L2421>

- Fix wrong prop names in `ComputedSlideSize`

- Add `RevealState` for `getState` `setSate` api

- Correct `getSlide` api

- Correct `getSlideBackground` api

- Correctly mark all optional params

- Changed all `Element` to `HTMLElement` since reveal.js heavily relies on DOM, all elements should come from browser or mocking env.

- Typo fixes

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/reveal.js/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a7345104b641d807c1a234dc1527e34634b1e1bf
</p>

**Comment by Mister-Hope** [comment]
>🔔 @robertop87 @Nava2 @JPtenBerge @kwatanwa17

File: types/reveal.js/index.d.ts [hunk 1]
```diff
@@ -1519,10 +1534,14 @@ declare namespace Reveal {
         tokenizer?: object;
         walkTokens?: (...args: any) => any;
         xhtml?: boolean;
+        separator?: string;
```
Comment (gabritto) [inline types/reveal.js/index.d.ts:1537]: Can you point to where in the original library this is documented, or used in the source code?

**Review by gabritto — CHANGES_REQUESTED** [review]
Looks good except for one of the changes that I couldn't verify; please clarify the change I commented on and I can approve this.

File: types/reveal.js/index.d.ts (hunk 1 above)
Comment (Mister-Hope) [reply types/reveal.js/index.d.ts:1537]: https://github.com/hakimel/reveal.js/blob/bbd0d3e4f73cf48a2062cacc4f7da3c25641c408/plugin/markdown/plugin.js#L101

**Review by gabritto — APPROVED** [review]

**Comment by Mister-Hope** [comment]
Ready to merge

**Pull request merged**